### ISSUE TITLE
testsuite does not pass when parallelized

### ISSUE THREAD

**Issue by pemensik** [yhirose/cpp-httplib#1494]
I made a package of cpp-httplib running also tests on Fedora. Our ``%ctest`` RPM macro expands to:
```
/usr/bin/ctest --test-dir "redhat-linux-build" --output-on-failure --force-new-ctest-process -j4
```

When used with -j1, all tests pass fine. But with multiple tests run at the same time, it fails reliably in multiple tests. It would be nice if some simple change could order tests to wait for conflicting tests, but would allow at least some tests in parallel. Test suite takes relatively long time.

It is just low severity issue, I made a simple workaround by using ``%ctest -j1``. Just nice to have. If you knew simple changes to fix that, it would be nice. On my quad core machine it fails reliably when manually run with ``ctest -j4``.

**Comment by yhirose** [comment]
@jimmy-park, it seems like another issue with the CTest support. Could you take a look at it? Thanks!

**Comment by jimmy-park** [comment]
This is not an issue of CTest itself. Many test cases are not suitable for running in parallel since they try to bind the same port which is hardcoded `1234`. Maybe replacing `listen()` to `bind_to_any_port() && listen_after_bind()` helps this issue?

**Comment by yhirose** [comment]
@jimmy-park, thanks for the replay.
@pemensik, it's not trivial changes to assign a different port to each unit test. So, I'll close it. Thanks for your understanding.
